### PR TITLE
three changes:

### PR DESCRIPTION
**Pull request by derekdreery**
- Add critical sections where there is a chance that an interrupt can
  be missed.
- Add blocking versions of SPIM methods.
- Make some pins on SPIM optional.

### CODE REVIEW

File: embassy-nrf/src/spim.rs [hunk 1]
```diff
@@ -184,19 +253,20 @@ impl<'d, T: Instance> FullDuplex<u8> for Spim<'d, T> {
...
+                        return Poll::Ready(());
+                    }
+
+                    wake_on_interrupt(&mut this.irq, cx.waker());
+                    Poll::Pending
+                })
             })
             .await;
 
+            compiler_fence(Ordering::SeqCst);
```
Comment (derekdreery) [inline embassy-nrf/src/spim.rs:269]: might not be needed

**Comment by Dirbaio** [comment]
I believe the race is fixed by 28664406d660c635a79fe71dfa7b019ea0228600 and 1dcffe0eea66687428e173ee2b958acde51019ea. It'd be awesome if you could give it a quick test! :)

**Comment by derekdreery** [comment]
I would test it but I fried my board :'(

**Comment by Dirbaio** [comment]
The race condition is fixed, the optional SPIM pins change has been included.

The only thing missing is blocking APIs, tracked in #136 . I agree it's something that we want but ideally it should be implementing the `embedded_hal` traits.

**Comment by derekdreery** [comment]
Cool I'm looking forward to trying this out once I've got my new hardware :)

**Comment by derekdreery** [comment]
Finally got some new hardware to try on. I think I'm still getting the race condition: my code stopped working when I went back to using embassy master. Will try to do some more digging.

**Pull request closed without merging**